src/planwise/component/datasets2.clj

```diff
@@ -197,13 +192,25 @@
   (let [db-spec   (get-db store)
         region-id (:region-id filter-options)
         algorithm (:coverage-algorithm filter-options)
-        options   (:coverage-options filter-options)]
+        options   (:coverage-options filter-options)
+        tags      (str/join "& " (:tags filter-options))]
     (db-find-sites-with-coverage-in-region db-spec {:dataset-id dataset-id
                                                     :version    version
                                                     :region-id  region-id
                                                     :algorithm  algorithm
-                                                    :options    (some-> options pr-str)})))
-
+                                                    :options    (some-> options pr-str)
+                                                    :tags tags})))
+
+(defn count-sites-filter-by-tag
+  [store dataset-id tag]
+  (let [db-spec  (get-db store)
+        count-fn (fn [db-spec tag]
```

Comment:
Since the `count-fn` is defined in the let binding, you don't need to pass `db-spec` as an argument. It is included in the closure scope.